Prepare release 3.4.0

sergei-maertens opened the pull request:
Closes #5830 - Depends on https://github.com/open-formulieren/open-forms-sdk/pull/929

[skip: e2e]

---

**Changes**

* Bumped version numbers
* Updated docs about versioning policy
* Updated Dutch translations
* Added changelog entry

**Checklist**

Check off the items that are completed or not relevant.

- Impact on features

  - [x] Checked copying a form
  - [x] Checked import/export of a form
  - [x] Config checks in the configuration overview admin page
  - [x] Problem detection in the admin email digest is handled

- Dockerfile/scripts

  - [x] Updated the Dockerfile with the necessary scripts from the `./bin` folder

- Commit hygiene

  - [x] Commit messages refer to the relevant Github issue
  - [x] Commit messages explain the "why" of change, not the how

SonnyBA commented on docs/developers/versioning.rst:
```diff
@@ -73,10 +73,10 @@ Version         Release date    API specification
 ==============  ==============  =============================
 latest          n/a             `ReDoc <https://redocly.github.io/redoc/?url=https://raw.githubusercontent.com/open-formulieren/open-forms/main/src/openapi.yaml>`__,
                                 `Swagger <https://petstore.swagger.io/?url=https://raw.githubusercontent.com/open-formulieren/open-forms/main/src/openapi.yaml>`__
+3.4.0           2026-01-05      `ReDoc <https://redocly.github.io/redoc/?url=https://raw.githubusercontent.com/open-formulieren/open-forms/3.4.0/src/openapi.yaml>`__,
```
Not sure how strict we are with these dates but if we are this should be updated before merging

SonnyBA commented on docs/introduction/versions.rst:
```diff
@@ -4,38 +4,59 @@
+calendar year - we do this to minimize the impact. Minor versions may deprecate
+functionalities that are then usually removed in the first upcoming major version.
+
+Next to that, we aim to publish a monthly patch release with bugfixes, for all supported
```
We could add a link here linking to the supported versions although it won't be very hard currently to miss that section.

sergei-maertens replied:
I think it's a bit overkill for now, as clicking it would just result in a page shift where users might expect a different page to open entirely.

sergei-maertens commented:
> Changelog entries for `#5736`, `#5817`, `#5830`, `#5813`, `#5833`, `#5831` are missing but I think they were excluded with (good) reason. 

Yep, those are just not relevant for the readers of the release notes or fit in the bigger entry (e.g. "we added Open Klant support", it's a bit silly to also mention "we tested it and fixed issues we discovered").

> I don't see an entry though for `#5762` ([see one of the commits](161bc2df0334ecd8d7f67572a79d35501b3db877)). 

This too is part of "we added support for Open Klant and tried to make the admin UI as helpful as possible". Those should be givens and be part of delvering a feature. It may be interesting for developers, but we have the git commit history that should provide sufficient information.

> Also think `#5831` and `#5729`, `#5876` are worth mentioning.

I strongly disagree - the email digest aspect is part of the PR checklist. If this is not covered, the feature is not finished.

For the KVK support: it's just part of the feature as is, and covered in the documentation that it's supported. It's also highlighted as a supported authentication flavour in the changelog, just without issue reference.

The updating fix if there's an empty address -> that was just a bug in an unreleased (to stable) feature. The assumption when releasing new features is that it works properly, bugs for in-development versions are just not interesting to the audience reading the changelog and checking what new features are available and which bugs are fixed, compared to 3.3.0.
